src/storage/StorageMessageQueue.ts

```diff
@@ -96,6 +98,11 @@ export class StorageServiceMessageQueue implements MessagePickupRepository {
       await this.sendNotification(this.agentContext, connectionId, 'messageType')
     }
 
+    // Send a Firebase Cloud Message notification to the device
+    if (config.get('agent:usePushNotifications') && process.env.FIREBASE_PROJECT_ID) {
+      await this.sendFCMNotification(this.agentContext, connectionId)
```

Comment:
I'm not sure if it's worth to create a separate conditional here or if you can concentrate all the logic within `this.sendNotification()`. I mean:

```ts
 if(config.get('agent:usePushNotifications`) {
      await this.sendNotification(this.agentContext, connectionId, 'messageType')
}
```

And inside `sendNotification` or you can check for other environment variables (`agent:notificationWebhookUrl`, `FIREBASE_PROJECT_ID`)  to determine if it should send FCM notification through internal sender and/or the external one (i.e. the one located at notification webhook URL).